pkg/ccl/backupccl/backup_job.go

```diff
@@ -809,10 +814,17 @@ func (b *backupResumer) maybeNotifyScheduledJobCompletion(
 }
 
 // OnFailOrCancel is part of the jobs.Resumer interface.
-func (b *backupResumer) OnFailOrCancel(ctx context.Context, execCtx interface{}) error {
+func (b *backupResumer) OnFailOrCancel(
+	ctx context.Context, execCtx interface{}, jobErr error,
+) error {
 	telemetry.Count("backup.total.failed")
 	telemetry.CountBucketed("backup.duration-sec.failed",
 		int64(timeutil.Since(timeutil.FromUnixMicros(b.job.Payload().StartedMicros)).Seconds()))
+	eventType := backupJobEventType
+	if b.job.Details().(jobspb.BackupDetails).ScheduleID != 0 {
+		eventType = scheduledBackupJobEventType
+	}
+	logJobCompletion(ctx, eventType, b.job.ID(), jobs.StatusFailed, jobErr)
```

Comment:
Is `StatusFailed` what we want if we were cancelled?

🤔 I kinda wonder if the job system should actually do the logging here when a job enters a terminal state, with some method on the Resumer interface that would return information that the job wants to log in such events.  Probably not worth doing until we see if another job ends up with similar looking code.

Reply:
Now I'm using `jobs.HasErrJobCanceled()` to determine if the job failure is because of a cancel.